### PR TITLE
DA-4596: dto changes for caching auth0 jwks

### PR DESCRIPTION
**Pull request by ajinkyan83**
This jwks certificate will auto-refresh & updated in the cache, for below 2 cases 
A) whenever the token is generated via the auth0 refresh token lambda OR
B) If the cert is requested for and if the cached jwks cert has expired (timeout set to 30 mins).

The cached jwks cert will be used whenever the consuming clients request for an access token to validate the same.

**Testing Done:**

Executed da-metrics/cron/auth0 locally on test env.
`auth0-jwks-cache-prod` and `auth0-jwks-cache-test` are created with 1 document with `_id` : `jwks`

Signed-off-by: Ajinkya Nahar <ajinkyan@proximabiz.com>

### CODE REVIEW

File: auth0/dto.go [hunk 1]
```diff
@@ -79,12 +116,17 @@ const (
...
+	auth0JwksCache        = "auth0-jwks-cache-"
+	jwksDoc               = "jwks"
 )
 
 // RefreshResult ...
 type RefreshResult string
 
-const(
+// RefreshResultJwks ...
+type RefreshResultJwks string
```
Comment (khalifapro) [inline auth0/dto.go:127]: Not good to have a type from type unless the new type does receive a new functionalities, reference for that is Duration type which is int64 in origin

**Review by khalifapro — APPROVED** [review]

File: auth0/dto.go (hunk 1 above)
Comment (ajinkyan83) [reply auth0/dto.go]: Addressed!

**Review by linonymous — APPROVED** [review]

**Review by enkhalifapro — CHANGES_REQUESTED** [review]
Regarding `auth0JwksCache` creation, it should be done in another cron job instead of having it when `getPemCert`

**Comment by ajinkyan83** [comment]
> Regarding `auth0JwksCache` creation, it should be done in another cron job instead of having it when `getPemCert`
@enkhalifapro @khalifapro 
I did give a though to this approach during implementation of the logic.
In this case, I will need to implement the refresh methods specific for jwks cache right, where I will need to fetch the token string from cache and Parse it to get the token object as done in this code [here ](https://github.com/LF-Engineering/dev-analytics-libraries/blob/main/auth0/token.go#L226)-  which in turn means the same step. So, I feel this approach is better instead of having another cron job.

Let me know if you agree, else we can sure have a discussion post standup.

**Comment by ajinkyan83** [comment]
@khalifapro As discussed in the call, merging this PR.

**Pull request merged**